### PR TITLE
[mmptest] Port link-uithread-1/2 to an NUnit LinkAll test. Partial fix for #4975.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Fixes part of https://github.com/xamarin/xamarin-macios/issues/4975.

### CODE REVIEW

File: tests/linker/mac/link all/LinkAllTest.cs [hunk 1]
```diff
@@ -25,5 +29,41 @@ public void Calendars ()
...
+			// works on main/ui thread
+			NSApplication.EnsureUIThread ();
+
+			ThreadPool.QueueUserWorkItem ((v) => Tester.Test ());
+			Assert.IsTrue (Tester.mre.WaitOne (TimeSpan.FromSeconds (10)), "Successful wait");
+#if DEBUG
+			var expected_ex_thrown = true;
+#else
+			var expected_ex_thrown = false;
+#endif
```
Comment (mandel-macaque) [inline tests/linker/mac/link all/LinkAllTest.cs]: Approving, but @rolfbjarne could we have a comment of why is this added? I'm sure we will forget

Comment (rolfbjarne) [reply tests/linker/mac/link all/LinkAllTest.cs]: It's because the UI thread check exception only occur in debug builds; in release builds the linker removes the check.
